src/api/app/controllers/webui/packages/badge_controller.rb

```diff
@@ -15,7 +15,7 @@ def show
 
       def create_badge(results)
         filename = 'badge-unknown.svg'
-        return Rails.application.assets[filename].source if results.blank?
+        return assets_file(filename) if results.blank?
```

Comment:
I cannot comment on already commited code, but it's much better if this function does not access the `params` hash in line 22: it will be easier to change in the future if the code in this function consumes local variables only.